include/IPipelineManager.hpp

```diff
@@ -30,15 +26,14 @@ namespace api
              * @brief Stops the pipeline listener when called. If the listener is already stopped, it should throw an
              * exception.
              */
-            void stop() = 0;
+            virtual void stop() = 0;
 
-        protected:
             /**
-             * @brief Handles a client request by calling the parser, the modules and the sender.
-             * @param packet[in] Refers to the packet that was sent by the client.
-             * @param client[in] Refers to the client that sent the packet.
+             * @brief Configures the pipeline by loading modules, parsers, senders, listeners.
+             * @param config[in] The PipelineConfig struct containing all paths to libraries.
+             * @warning If the PipelineManager is unable to load the library, it should throw an exception.
```

Comment:
why not use @throw ?

Reply:
with @throw i MUST explain which exception will be thrown and it's the user choice